modules/rostests/apitests/advapi32/RegCreateKeyEx.c

```diff
@@ -0,0 +1,186 @@
+    if (RegOpenKeyExW(HKEY_CURRENT_USER, L"mykey", 0, KEY_ALL_ACCESS, &hkey_main) == ERROR_SUCCESS)
+    {
+        RegCloseKey(hkey_main);
+        RegDeleteKeyExW(HKEY_CURRENT_USER, L"mykey", KEY_ALL_ACCESS, 0);
+    }
+
+    if (RegOpenKeyExW(HKEY_CURRENT_USER, L"mykey1", 0, KEY_ALL_ACCESS, &hkey_main) == ERROR_SUCCESS)
+    {
+        RegCloseKey(hkey_main);
+        RegDeleteKeyExW(HKEY_CURRENT_USER, L"mykey1", KEY_ALL_ACCESS, 0);
+    }
+
+    if (RegOpenKeyExW(HKEY_CURRENT_USER, L"mykey2", 0, KEY_ALL_ACCESS, &hkey_main) == ERROR_SUCCESS)
+    {
+        RegCloseKey(hkey_main);
+        RegDeleteKeyExW(HKEY_CURRENT_USER, L"mykey2", KEY_ALL_ACCESS, 0);
+    }
+
+    /* Setup GetLastError to known value for tests */
+    SetLastError(0xdeadbeef);
+
+    // Create a well-known SID for the Everyone group.
+    bRes = AllocateAndInitializeSid(&SIDAuthWorld, 1,
+           SECURITY_WORLD_RID,
+           0, 0, 0, 0, 0, 0, 0,
+           &pEveryoneSID);
+    ok(bRes == TRUE, "AllocateAndInitializeSid Error %ld\n", GetLastError());
+    if (bRes != TRUE)
+        goto Cleanup;
+
+    // Initialize an EXPLICIT_ACCESS structure for an ACE.
+    // The ACE will allow Everyone read access to the key.
+    ZeroMemory(&ea, sizeof(ea));
+    ea[0].grfAccessPermissions = KEY_READ;
+    ea[0].grfAccessMode = SET_ACCESS;
+    ea[0].grfInheritance= NO_INHERITANCE;
+    ea[0].Trustee.TrusteeForm = TRUSTEE_IS_SID;
+    ea[0].Trustee.TrusteeType = TRUSTEE_IS_WELL_KNOWN_GROUP;
+    ea[0].Trustee.ptstrName  = pEveryoneSID;
+
+    // Create a SID for the BUILTIN\Administrators group.
+    bRes = AllocateAndInitializeSid(&SIDAuthNT, 2,
+        SECURITY_BUILTIN_DOMAIN_RID,
+        DOMAIN_ALIAS_RID_ADMINS,
+        0, 0, 0, 0, 0, 0,
+        &pAdminSID);
+    ok(bRes == TRUE, "AllocateAndInitializeSid Error %ld\n", GetLastError());
+    if (bRes == FALSE)
+        goto Cleanup;
+
+    // Initialize an EXPLICIT_ACCESS structure for an ACE.
+    // The ACE will allow the Administrators group full access to the key.
+    ea[1].grfAccessPermissions = KEY_ALL_ACCESS;
+    ea[1].grfAccessMode = SET_ACCESS;
+    ea[1].grfInheritance= NO_INHERITANCE;
+    ea[1].Trustee.TrusteeForm = TRUSTEE_IS_SID;
+    ea[1].Trustee.TrusteeType = TRUSTEE_IS_GROUP;
+    ea[1].Trustee.ptstrName  = pAdminSID;
+
+    // Create a new ACL that contains the new ACEs.
+    dwRes = SetEntriesInAclW(_countof(ea), ea, NULL, &pACL);
+    ok(dwRes == ERROR_SUCCESS, "SetEntriesInAcl Error %ld\n", GetLastError());
+    if (dwRes != ERROR_SUCCESS)
+        goto Cleanup;
+
+    // Initialize a security descriptor.
+    pSD = LocalAlloc(LPTR, SECURITY_DESCRIPTOR_MIN_LENGTH);
+    ok(pSD != NULL, "LocalAlloc Error %ld\n", GetLastError());
+    if (pSD == NULL)
+        goto Cleanup;
+ 
+    bRes = InitializeSecurityDescriptor(pSD, SECURITY_DESCRIPTOR_REVISION);
+    ok(bRes == TRUE, "InitializeSecurityDescriptor Error %ld\n", GetLastError());
+    if (!bRes)
```

Comment:
I don't see any benefit in this proposal.

Reply:
ThFabba noticed such issues. Doug-Lyons fixed that partially. I am simply pointing out remaining cases here and elsewhere.

Reply:
What do you mean with "such issues"? The code-flow is 100% the same before and after your change, isn't it?
All I can see is that your proposal uses more characters than before to achieve the same. So I can't see any improvement here. If anybody else thinks the same as me, then please resolve those suggestions without further action.

Reply:
I mean https://github.com/reactos/reactos/pull/5234#discussion_r1161276904 .
I am more interested in code/style consistency than in number of characters.

Reply:
"function returns non-zero on success" makes this kind of plausible, ok.
I have not looked up the ok-macro yet, but very likely it is also possible to just write
ok(bRes, ...
instead of
ok(bRes != FALSE, ...

and ok(!bRes, ...
instead of
ok(bRes == FALSE, ...

But I guess, that is just a matter of taste. Feel free to go your preferred way Doug.

Reply:
After re-reading Thomas Faber's comments, I have reviewed the returned values for the functions and changed the tests accordingly. Thanks.